lib/openstack_query/mappings/flavor_mapping.py

```diff
@@ -27,6 +29,22 @@ class FlavorMapping(MappingInterface):
     Define property mappings, kwarg mappings and filter function mappings, and runner mapping related to flavors here
     """
 
+    @staticmethod
+    def get_chain_mappings():
+        """
+        Should return a dictionary containing property pairs mapped to query mappings.
+        This is used to define how to chain results from this query to other possible queries
+        """
+        # local import prevents circular import issue
+        from enums.query.query_types import QueryTypes
+
+        return {
+            QueryTypes.SERVER_QUERY: (
+                FlavorProperties.FLAVOR_ID,
```

Comment:
Do we need `QueryTypes.SERVER_QUERY`, surely we can do something like:

```
def get_chain_mappings():
    return {ServerProperties.FLAVOR_ID: FlavourProperties.FLAVOR_ID}


# Chaining module
original_properties = query_output.get_props()
mappings = flavor.get_chain_mappings()
found_key = next(prop for prop in original_properties if prop in mappings, None)

if found_key:
   matching_key = mappings[found_key]
```

This only has to be written once by the chaining module, and it completely decouples property mappings from QueryTypes, since we don't really care at this level

This means that the chain_mapping in all the classes don't bloat with having to know what other query types link in too

Reply:
it will have to be instead
```
found_key = next(prop for prop in original_properties if prop.name in mappings, None)
```
The distinction here is `prop.name` - we create a implicit equality here that the **names** should match between enums 

This is something I'd rather do explicitly - i.e. mappings the way I set out.

Reply:
Why do we need to prop name? Since the existing query will have a `ServerProperties.FLAVOR_ID` set somewhere in the select for it to work 

We just need to retrieve the keys passed to the select?

Reply:
Ignore - I was misunderstanding - I'm happy with this change :+1:

Reply:
Also separately to this, if you don't do a `select` does that implicitly put all the `ServerProperties` enum fields into a n internally tracked list of properties or similar? As that would be a requirement of this